modules/caddyhttp/matchers.go

```diff
@@ -909,25 +914,51 @@ func (m MatchRemoteIP) getClientIP(r *http.Request) (net.IP, error) {
 	if err != nil {
 		ipStr = remote // OK; probably didn't have a port
 	}
+	// Some IPv6-Adresses can contain zone identifiers at the end,
+	// which are separated with "%"
+	if strings.Contains(ipStr, "%") {
+		zone_id = strings.Split(ipStr, "%")[1]
+		ipStr = strings.Split(ipStr, "%")[0]
+	}
 	ip := net.ParseIP(ipStr)
 	if ip == nil {
-		return nil, fmt.Errorf("invalid client IP address: %s", ipStr)
+		return nil, zone_id, fmt.Errorf("invalid client IP address: %s", ipStr)
 	}
-	return ip, nil
+	return ip, zone_id, nil
 }
 
 // Match returns true if r matches m.
 func (m MatchRemoteIP) Match(r *http.Request) bool {
-	clientIP, err := m.getClientIP(r)
+	clientIP, zone_id_got, err := m.getClientIP(r)
+	zoneFilter := true
 	if err != nil {
 		m.logger.Error("getting client IP", zap.Error(err))
 		return false
 	}
```

Comment:
This variable relates to the loop, not to the `if err != nil`, so it should go after.

```suggestion
	if err != nil {
		m.logger.Error("getting client IP", zap.Error(err))
		return false
	}
	zoneFilter := true
```

Reply:
makes sense